packages/html-reporter/src/testResultView.tsx

```diff
@@ -161,19 +170,23 @@ function classifyErrors(testErrors: string[], diffs: ImageDiff[]) {
 }
 
 const StepTreeItem: React.FC<{
+  test: TestCase;
+  result: TestResult;
   step: TestStep;
   depth: number,
-}> = ({ step, depth }) => {
-  return <TreeItem title={<span>
+}> = ({ test, step, result, depth }) => {
+  const attachmentName = step.title.match(/^attach "(.*)"$/)?.[1];
+  return <TreeItem title={<span aria-label={step.title}>
     <span style={{ float: 'right' }}>{msToString(step.duration)}</span>
+    {attachmentName && <a style={{ float: 'right' }} title='link to attachment' href={`#?testId=${test.testId}&anchor=attachment-${encodeURIComponent(attachmentName)}&run=${test.results.indexOf(result)}`} onClick={evt => { evt.stopPropagation(); }}>{icons.attachment()}</a>}
```

Comment:
- I wonder why do we use `<Link>` in some places, and plain `<a>` in others?
- Can we simplify link generation to avoid passing `test` and `result` around somehow? Perhaps take existing ones from the current url? Or put them into a context? Something for a followup.

Reply:
- no idea 🤷 I guess I can also rewrite this to `<Link>`, but I don't think there'll be a big difference since it's not a text link but a button
- agree, that'd be nice. i'll look at it in a followup.